DigitalOffice/Services/TimeManagementService/Startup.cs

```diff
@@ -0,0 +1,58 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using TimeManagementService.Database;
+
+namespace TimeManagementService
+{
+    public class Startup
+    {
+        public IConfiguration Configuration { get; }
+
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<TimeManagementDbContext>(options =>
+            {
+                options.UseSqlServer(Configuration.GetConnectionString("SQLConnectionString"));
+            });
+
+            services.AddControllers();
+        }
+
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            UpdateDatabase(app);
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+        }
+
+        private static void UpdateDatabase(IApplicationBuilder app)
```

Comment:
why static?

Reply:
Maybe static method used, because this allows call method without having to create specific instance of that class.

Reply:
Private method ?????

Reply:
This is yes), this will not work with a private method, there will be no way to access the method from outside.

Reply:
@DarkFelas  "static" allows the compiler to skip implicit "this." pointer and we make sure that this method can't change Startup's object state, etc

Reply:
> @DarkFelas "static" allows the compiler to skip implicit "this." pointer and we make sure that this method can't change Startup's object state, etc

We need precise performance tests at this point :) im not sure that in this case it calls "this" multiple times

Reply:
Hm, not sure what does it mean in this context. What state of Startup object can be changed and why it is bad?

Reply:
If we're talking about the state of Startup object, since this is a private method we won't be able to use this method elsewhere, that's true.
Also true is that usually we don't create objects of Startup. So these arguments may be put aside.

So my only argument is that we make a teeny-tiny shortcut, as I said previously, to skip implicit "this." pointer. Other than that I dont have anything to add about the topic.